### PR TITLE
use sign from token if present

### PR DESCRIPTION
**Pull request by yCobanoglu**

### CODE REVIEW

File: src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+} from 'transliteration/domain/token'
+import { AnnotationTokenType } from 'fragmentarium/domain/annotation'
+import { isNamedSign } from 'transliteration/domain/type-guards'
+import { AbstractLine } from 'transliteration/domain/abstract-line'
+import { SurfaceAtLine } from 'transliteration/domain/at-lines'
+import { RulingDollarLine } from 'transliteration/domain/dollar-lines'
+import { Text } from 'transliteration/domain/text'
+import { AnnotationToken } from 'fragmentarium/domain/annotation-token'
+
+function namedSignTokenToAnnotationToken(
```
[the remark this answers is not in the document]
Comment (jlaasonen) [reply src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.ts:16]: You could extract a method for the deciding `type`.

File: src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  } else if (token.parts) {
+    return token.parts.flatMap((part: Token, index: number) =>
+      mapToken(part, [...path, index])
+    )
+  } else {
+    return unannotatableTokenToAnnotationToken(token, path)
+  }
+}
+
+function structureLineToTokens(
```
[the remark this answers is not in the document]
Comment (jlaasonen) [reply src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.ts:121]: You could extract methods for handling each type of line.

File: src/signs/infrastructure/SignRepository.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ class SignRepository {
   private attachSignToToken(
     token: AnnotationToken
   ): Promise<AnnotationToken> | AnnotationToken {
-    if (token.isSignPossiblyExisting()) {
+    if (token.isSignPossiblyExisting() && token.sign == null) {
```
Comment (jlaasonen) [inline src/signs/infrastructure/SignRepository.ts:32]: Use strict equality `===`. Consider adding `hasSign` to `AnnotationToken`.

**Review by jlaasonen — CHANGES_REQUESTED** [review]

File: src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.test.ts [hunk 4]
```diff
@@ -164,9 +192,9 @@ it('', () => {
       suffixModifier: null,
       type: 'LineNumber',
     },
-  })
+  }
   const text = new Text({
-    lines: [at.surface, textLine, dollar.singleRuling],
+    lines: [at.surface, (textLine as unknown) as TextLine, dollar.singleRuling],
```
Comment (jlaasonen) [inline src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.test.ts:197]: Do not do this. `textLine` is not a `TextLine`. Not sure what you are after? It was a `TextLine`. You changed it, but cast (incorrectly) it back to a `TextLine`

**Review by jlaasonen — CHANGES_REQUESTED** [review]

File: src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.test.ts (hunk 4 above)
Comment (yCobanoglu) [reply src/fragmentarium/ui/image-annotation/annotation-tool/mapTokensToAnnotationTokens.test.ts]: i changed the token interface

File: src/transliteration/domain/token.ts [hunk 5]
```diff
@@ -111,10 +111,15 @@ export interface NamedSign extends Sign {
   readonly name: string
   readonly nameParts: readonly (ValueToken | Enclosure)[]
   readonly subIndex?: number | null
-  readonly sign?: Token | null
+  readonly sign?: Token | null | CompoundGraphemeSign
```
Comment (jlaasonen) [inline src/transliteration/domain/token.ts:114]: This is redundant `CompoundGraphemeSign` is a `Token`.

File: src/transliteration/domain/token.ts [hunk 6]
```diff
@@ -111,10 +111,15 @@ export interface NamedSign extends Sign {
...
   readonly surrogate?: readonly Token[] | null
 }
 
+interface CompoundGraphemeSign extends BaseToken {
```
Comment (jlaasonen) [inline src/transliteration/domain/token.ts:118]: What's this for? There is already `CompoundGrapheme`.

**Review by jlaasonen — CHANGES_REQUESTED** [review]

**Review by jlaasonen — APPROVED** [review]

**Pull request merged**